openmc/tallies.py

```diff
@@ -1410,6 +1432,765 @@ def export_results(self, filename='tally-results', directory='.',
+            other_tile_factor = num_score_bins / other_num_score_bins
+
+            # Replicate the data
+            self_mean = np.repeat(self_mean, self_repeat_factor, axis=2)
+            other_mean = np.tile(other_mean, (1, 1, other_tile_factor))
+            self_std_dev = np.repeat(self_std_dev, self_repeat_factor, axis=2)
+            other_std_dev = np.tile(other_std_dev, (1, 1, other_tile_factor))
+
+        data = {}
+        data['self'] = {}
+        data['other'] = {}
+        data['self']['mean'] = self_mean
+        data['other']['mean'] = other_mean
+        data['self']['std. dev.'] = self_std_dev
+        data['other']['std. dev.'] = other_std_dev
+        return data
+
+    def __add__(self, other):
+        """Adds this tally to another tally or scalar value.
+
+        This method builds a new tally with data that is the sum of this
+        tally's data and that from the other tally or scalar value. If the
+        filters, scores and nuclides in the two tallies are not the same, then
+        they are combined in all possible ways in the new derived tally.
+
+        Uncertainty propagation is used to compute the standard deviation
+        for the new tally's data. It is important to note that this makes
+        the assumption that the tally data is independently distributed.
+        In most use cases, this is *not* true and may lead to under-prediction
+        of the uncertainty. The uncertainty propagation model is from the
+        following source:
+
+        https://en.wikipedia.org/wiki/Propagation_of_uncertainty
+
+        Parameters
+        ----------
+        other : Tally or Integer or Rational
+            The tally or scalar value to add to this tally
+
+        Returns
+        -------
+        Tally
+            A new derived tally which is the sum of this tally and the other
+            tally or scalar value in the addition.
+
+        """
+
+        # Check that results have been read
+        if self.sum is None:
+            msg = 'Unable to use tally arithmetic with Tally ID={0} ' \
+                  'since it does not contain any results.'.format(self.id)
+            raise ValueError(msg)
+
+        new_tally = Tally()
+        new_tally.with_batch_statistics = True
+
+        if isinstance(other, Tally):
+
+            # Check that results have been read
+            if other.sum is None:
+                msg = 'Unable to use tally arithmetic with Tally ID={0} ' \
+                      'since it does not contain any results.'.format(other.id)
+                raise ValueError(msg)
+
+            self._outer_product(other, new_tally, binary_op='+')
+            data = self._align_tally_data(other)
+            new_tally._mean = data['self']['mean'] + data['other']['mean']
+            new_tally._std_dev = np.sqrt(data['self']['std. dev.']**2 + \
+                                         data['other']['std. dev.']**2)
+
+        elif isinstance(other, (Integral, Rational)):
```

Comment:
Whether Rational or Real is the right choice, you shouldn't need the check against Integral since integers are reals/rationals.